drivers/ipm/ipm_cavs_idc.h

```diff
@@ -7,6 +7,26 @@
 #ifndef ZEPHYR_DRIVERS_IPM_IPM_CAVS_IDC_H_
 #define ZEPHYR_DRIVERS_IPM_IPM_CAVS_IDC_H_
 
+/* Redeclaration of the earlier IDC register API for platforms being
+ * held back on this driver.
+ */
+#ifndef CONFIG_SOC_SERIES_INTEL_CAVS_V25
```

Comment:
As of this PR, the old driver is symmetric and "can" be used, so I figured I might as well declare it that way.  The reason for specifying 2.5 specifically is that it could never use IPM (which limits us to 2 cores) and is limited to the new driver only.